services/navigation/builtin/builtin.go

```diff
@@ -575,7 +581,75 @@ func (svc *builtIn) waypointIsDeleted() bool {
 func (svc *builtIn) Obstacles(ctx context.Context, extra map[string]interface{}) ([]*spatialmath.GeoObstacle, error) {
 	svc.mu.RLock()
 	defer svc.mu.RUnlock()
-	return svc.obstacles, nil
+
+	// get static geoObstacles
+	geoObstacles := svc.obstacles
+
+	for _, detector := range svc.motionCfg.ObstacleDetectors {
+		// get the vision service
+		visSvc, ok := svc.visionServicesByName[detector.VisionServiceName]
+		if !ok {
+			return nil, fmt.Errorf("vision service with name: %s not found", detector.VisionServiceName)
+		}
+
+		// get the detections
+		detections, err := visSvc.GetObjectPointClouds(ctx, detector.CameraName.Name, nil)
```

Comment:
I feel like I've read this code before and that this exists already probably in the motion service?  If thats true it should be pretty simple to make a helper function as a single point of truth for getting the obstacles in both cases.

Reply:
hmmm, I see what you mean.. 
We have the `obstaclesIntersectPlan` method inside `move_request.go` which also uses a vision service to call `GetObjectPointClouds`, however, that is where the similarities between the two end imo.
The set-up to get detections within the motion service is entirely different than how it is handled within the navigation service, and the make them identical the `moverequest` would need a new field added to is.
For this reason, I am unsure if making a helper function as a single point of truth for getting the obstacles in both cases is useful.

Reply:
As we talked about offline, I don't know that we need to do this prior to merging this work, but some sort of TODO I do think is necessary.  Curious to hear what @nicksanford thinks on this too

Reply:
My vote is to not try to DRY up the code that is duplicated between motion & nav until we confirm that having Nav query the vision service independent of the motion service actually satisfies the use cases we want to solve.

In my opinion, the user of the nav service only cares about what transient obstacles motion can see & is going to act upon, not the obstacles the nav service can see (as nav doesn't change its behavior based on transient obstacles, while motion does).

The current API design assumes that the frontend polling the nav service asking what obstacles the vision service can see will return results that are close enough to what the motion service can see and is acting upon.  

I recommend we confirm that assumption holds up to hardware testing before investing in any DRYing up of the code.